File: core/src/commonMain/kotlin/work/socialhub/kmastodon/MastodonException.java
package work.socialhub.kmastodon;


import mastodon4j.entity.Error;

public class MastodonException extends RuntimeException {

    private Error error;
    private int statusCode;

    public MastodonException(Exception e) {
        super(e);
    }

    public MastodonException(Error error, int statusCode) {
        this.statusCode = statusCode;
        this.error = error;
    }

    public Error getError() {
        return error;
    }

    public int getStatusCode() {
        return statusCode;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/MastodonException.kt
package work.socialhub.kmastodon

class MastodonException : Exception {
    var status: Int? = null
    var body: String? = null

    constructor(m: String) : super(m)
    constructor(e: Exception) : super(e)

    constructor(
        status: Int,
        body: String
    ) : super("status code: $status, body: $body") {
        this.status = status
        this.body = body
    }
}
